sympy/simplify/cse_main.py

```diff
@@ -770,3 +843,71 @@ def cse(exprs, symbols=None, optimizations=None, postprocess=None,
         return replacements, reduced_exprs
 
     return postprocess(replacements, reduced_exprs)
+
+
+def _cse_homogeneous(exprs, **kwargs):
```

Comment:
I still think we don't need this function. The string support in lambdify does not need to be there, and I'm fine if it doesn't work when cse=True. The remaining cases should be handled in cse itself.

Reply:
> should be handled in cse itself.

not without breaking compatibility or adding a new keyword. `cse` always emits a list of expressions whether there is 1 expressions or more.

> lambdify does not need to be there

It is already there
```python
>>> lambdify(x,'x+1')(1)
2
>>> lambdify('x','x+1')(1)
2
>>> lambdify('x',x+1)(1)
2
```

Reply:
Most functions that accept string inputs do so by accident (although for whatever reason it's actually tested for lambdify). I'd like to deprecate and eventually remove implicit string parsing from SymPy functions (except for the handful where it actually makes sense, and lambdify isn't one of those).